operator/images/cluster-setup/content/bin/utils.sh

```diff
@@ -20,6 +20,35 @@ exit_error() {
   exit 1
 }
 
+check_applications() {
+  local ns="$1"
+  shift
+  local applications=("$@")
+
+  for app in "${applications[@]}"; do
+    printf -- "- %s: " "$app"
+
+    #a loop to check if the ArgoCD application exists
+    if ! timeout 300s bash -c "while ! kubectl get application/$app -n $ns >/dev/null 2>/dev/null; do printf '.'; sleep 10; done"; then
+      printf "%s not found (timeout)\n" "$app"
+      kubectl get application/"$app" -n "$ns"
+      kubectl -n "$ns" get events | grep Warning
+      exit 1
+    else
+      printf "Exists"
+    fi
+
+     #a loop to check if the ArgoCD application is synced
+    if kubectl wait --for=jsonpath="{.status.sync.status}"="Synced" "application/$app" -n "$ns" --timeout=200s >/dev/null; then
```

Comment:
over in infra-deployments they do something similar but slightly different:  https://github.com/redhat-appstudio/infra-deployments/blob/main/hack/preview.sh#L200-L203

first blush, I wonder if the distinction between `Synced` and `Healthy Synced` matters ... I myself would not claim to be knowledgeable enough to say

Reply:
Good catch - we need the health status as well: https://argo-cd.readthedocs.io/en/stable/operator-manual/health/